pkg/controller/endpointslice/utils_test.go

```diff
@@ -386,6 +501,170 @@ func TestGetEndpointPorts(t *testing.T) {
 	}
 }
 
+func TestSetEndpointSliceLabelsLabels(t *testing.T) {
+
+	service := v1.Service{
+		ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "test"},
+		Spec: v1.ServiceSpec{
+			Ports:     []v1.ServicePort{{Port: 80}},
+			Selector:  map[string]string{"foo": "bar"},
+			ClusterIP: "1.1.1.1",
+		},
+	}
+
+	testCases := []struct {
```

Comment:
I think this also needs some test cases for the IsHeadlessService label.

Reply:
I've added new ones but don't know exactly what are you looking for :)

Reply:
Thanks! You've mostly covered it here. I think one more that would be good to have would be setting the `IsHeadlessService` on a Service that is not actually headless and then ensuring that label is not set at all on any EndpointSlices.

Reply:
added 3 cases more for headless labels